src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/DuplicateReferenceAnalyzer.cs

```diff
@@ -23,16 +23,15 @@ public DuplicateReferenceAnalyzer(IVersionComparer comparer, ILogger<DuplicateRe
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         }
 
-        public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state, CancellationToken token)
+        public Task AnalyzeAsync(IProject project, IDependencyAnalysisState state, CancellationToken token)
         {
             if (state is null)
             {
                 throw new ArgumentNullException(nameof(state));
             }
 
             // If the package is referenced more than once (bizarrely, this happens one of our test inputs), only keep the highest version
-            var references = await project.Required().GetNuGetReferencesAsync(token).ConfigureAwait(false);
-            var packages = references.PackageReferences.ToLookup(p => p.Name);
+            var packages = project.Required().NuGetReferences.PackageReferences.ToLookup(p => p.Name);
```

Comment:
nit: i think we are getting rid of project.Required() and elevating as a guard clause

Reply:
my rule of thumb is that if there's more than one required, then we do the main guard. I'm happy to just do the guards as it seems we're having to rewrite this often. Should we get rid of the `Required()` method?